### DISCUSSION TITLE
Does _document's getInitialProps invalidate Auto Static Optimization?

### DISCUSSION THREAD

**Discussion by awareness481** [vercel/next.js#20482] [Help] [answered]
Building the `with-styled-components` example which uses `getInitialProps` in `_document.js` gives me this cli output

![image](https://user-images.githubusercontent.com/12380586/103147121-935e6d80-475a-11eb-812e-b705a0a49e1d.png)

Whilst the docs mention that `getInitialProps will disable Automatic Static Optimization.` In the [_document docs](https://nextjs.org/docs/advanced-features/custom-document)
it is mentioned that `Document's getInitialProps function is not called during client-side transitions, nor when a page is statically optimized.`.

In conclusion, although the build output suggests that the pages are static, I'm confused as to whether that's actually the case.

**Comment by awareness481** [comment]
*Bump*: I'd love to get an answer of whether `getInitialProps` in `_document.js` invalidates static optimization.

**Comment by lfades** [comment, marked as answer]
`_document` only runs in the server, and it doesn't affect static optimization.

**Reply by awareness481** [reply]
Thank you!! :1st_place_medal:

**Reply by adhsafal** [reply]
If getInitialProps is used in _document file, can we do SSG?
